src/components/UserAddModal.tsx

```diff
@@ -0,0 +1,105 @@
+import { User } from '@luna/api/auth/types';
+import {
+  Button,
+  Checkbox,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+} from '@nextui-org/react';
+import { useEffect, useState } from 'react';
+
+export interface UserAddModalProps {
+  show: boolean;
+  setShow: (show: boolean) => void;
+}
+
+export function UserAddModal({ show, setShow }: UserAddModalProps) {
+  const [user, setUser] = useState<User>({ username: '' });
+  const [password, setPassword] = useState('');
+
+  // initialize/reset modal state
+  useEffect(() => {
+    setUser({
+      username: '',
+    });
+    setPassword('');
+  }, [show]);
+
+  const addUser = () => {
+    const payload = {
+      username: user.username,
+      password,
+      email: user.email,
+      permanent_api_token: user.permanentApiToken,
+    };
+    console.log('adding user:', payload);
+    // TODO: call POST /users
+    // TODO: feedback from the request (success, error)
+    onOpenChange(false);
+  };
```

Comment:
I recommend wrapping all callbacks in `useCallback`, both for performance and to prevent unnecessary rerenders:

```suggestion
  const addUser = useCallback(() => {
    const payload = {
      username: user.username,
      password,
      email: user.email,
      permanent_api_token: user.permanentApiToken,
    };
    console.log('adding user:', payload);
    // TODO: call POST /users
    // TODO: feedback from the request (success, error)
    onOpenChange(false);
  }, [user, password]);
```